packages/ketcher-core/src/application/editor/Editor.ts

```diff
@@ -270,12 +270,18 @@ export class CoreEditor {
     this.unsubscribeEvents();
     const struct = this.micromoleculesEditor.struct();
     const reStruct = this.micromoleculesEditor.render.ctab;
-    MacromoleculesConverter.convertDrawingEntitiesToStruct(
-      this.drawingEntitiesManager,
-      struct,
-      reStruct,
-    );
+    const { conversionErrorMessage } =
+      MacromoleculesConverter.convertDrawingEntitiesToStruct(
+        this.drawingEntitiesManager,
+        struct,
+        reStruct,
+      );
     reStruct.render.setMolecule(struct);
+    if (conversionErrorMessage) {
+      global.ketcher.editor.setMacromoleculeConvertionError(
```

Comment:
I propose to use KetcherProvider here instead of `global`

Reply:
Done